embedded/document/document.go

```diff
@@ -99,3 +100,9 @@ func (d *Document) Get(field string) interface{} {
 	}
 	return nil
 }
+
+// mergeDocumentStructs merges the given document struct with the given document struct to update
+func mergeDocumentStructs(doc *structpb.Struct, toUpdateDoc *structpb.Struct) error {
+	proto.Merge(doc, toUpdateDoc)
```

Comment:
taking a quick look at `proto.Merge` it seems to be that the document to be updated is the first parameter.

First though is that merging may not be the expected behaviour for an update, something to review

Reply:
ack, fixed it